Source/Urho3D/UI/UIElement.cpp

```diff
@@ -220,6 +220,27 @@ bool UIElement::LoadXML(const XMLElement& source, bool setInstanceDefault)
 
 bool UIElement::LoadXML(const XMLElement& source, XMLFile* styleFile, bool setInstanceDefault)
 {
+    // If this is a root element get style file from which style is loaded.

```

Comment:
As commented before, I don't think it is wise to mix loading of UI-layout with UI-style file. The LoadXML() has assumed too much on how the style file could be loaded.

Reply:
Any idea what problems could this cause?

Reply:
What if I made a mistake to forget to load the style before loading the UI-layout and I want it to fail instead of conveniently auto-loading one from resource because I could have my style constructed from memory or by using any other means.

Reply:
Then your layout xml would not have a `styleFile` parameter and it would behave just like it does without this patch. It is totally optional.

Reply:
I am still not convince. We already have "defaultStyle_" member variable and now we also have "defaultStyleFileName_". So, which one is the ultimate source of truth. Have you looked at the implementation of the `UI::LoadLayout()`? If you modify both of these method overloads to optionally read the stylefile name attribute when it is available and use it to override the default style of the root element before calling the `UIElement::LoadXML()` then I am fine with that.

Reply:
I will remove this change. While something along these lines would be nice, there is no dire need of it. For now ill copy behaviour of current editor - pre-selecting default style before loading UI layout. We can revisit this matter in the future should there be a need.